src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.ChangeNamespaceCodeAction.cs

```diff
@@ -0,0 +1,482 @@
+        /// is global namespace, i.e. default namespace is "" and the file is located at project 
+        /// root directory, and no namespace declaration in the document, respectively.
+        /// </summary>
+        internal sealed class ChangeNamespaceCodeAction : CodeAction
+        {
+            private readonly State _state;
+            private readonly AbstractSyncNamespaceService<TNamespaceDeclarationSyntax, TCompilationUnitSyntax> _service;
+
+            public override string Title => _state.TargetNamespace.Length == 0 
+                ? FeaturesResources.Change_to_global_namespace
+                : string.Format(FeaturesResources.Change_namespace_to_0, _state.TargetNamespace);
+
+            public ChangeNamespaceCodeAction(
+                AbstractSyncNamespaceService<TNamespaceDeclarationSyntax, TCompilationUnitSyntax> service,
+                State state)
+            {
+                _service = service;
+                _state = state;
+            }
+
+            private ImmutableArray<string> _oldNamespaceParts;
+            private ImmutableArray<string> OldNamespaceParts
+            {
+                get
+                {
+                    if (_oldNamespaceParts.IsDefault)
+                    {
+                        _oldNamespaceParts = _state.DeclaredNamespace.Split(new[] { '.' }).ToImmutableArray();
+                    }
+                    return _oldNamespaceParts;
+                }
+            }
+
+            private ImmutableArray<string> _newNamespaceParts;
+            private ImmutableArray<string> NewNamespaceParts
+            {
+                get
+                {
+                    Debug.Assert(_state.TargetNamespace != null);
+
+                    if (_newNamespaceParts.IsDefault)
+                    {
+                        _newNamespaceParts = _state.TargetNamespace.Split(new[] { '.' }).ToImmutableArray();
+                    }
+                    Debug.Assert(_newNamespaceParts.Length > 0);
+                    return _newNamespaceParts;
+                }
+            }
+
+            private ImmutableArray<ISymbol> GetDeclaredSymbolsInContainer(
+                SemanticModel semanticModel, 
+                SyntaxNode node, 
+                CancellationToken cancellationToken)
+            {
+                var declarations = _service.GetMemberDeclarationsInContainer(node);
+                var builder = ArrayBuilder<ISymbol>.GetInstance();
+                foreach (var declaration in declarations)
+                {
+                    var symbol = semanticModel.GetDeclaredSymbol(declaration, cancellationToken);
+                    builder.AddIfNotNull(symbol);
+                }
+                return builder.ToImmutableAndFree();
+            }
+
+            protected override async Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(CancellationToken cancellationToken)
```

Comment:
this seems super weird.  we only override ComputeOperations if we need to do *very* specialized stuff.

Reply:
Changed to overriding `Task<Solution> GetChangedSolutionAsync(IProgressTracker progressTracker, CancellationToken cancellationToken)`